# Data generation

Importing libreries

In [1]:
import pandas as pd
import seaborn as sns
import numpy as np

## For sheet_1 , Patient Record

import dataset format

In [2]:
dt=pd.read_excel("DataSets/Dataset Formats.xlsx",sheet_name="Patient Record")
dt.head()

Sr. No. Patient Id  Name of Patient  Adhar ID  DOB   Parent ID  \
0        1  RajMah001              NaN       NaN  NaN  RajMahP001   
1        2  RajMah002              NaN       NaN  NaN  RajMahP002   
2        3  RajMah003              NaN       NaN  NaN  RajMahP003   
3        4  RajMah004              NaN       NaN  NaN  RajMahP004   
4        5  RajMah005              NaN       NaN  NaN  RajMahP005   

   Contact No.  Address  Unnamed: 8                     Unnamed: 9  
0          NaN      NaN         NaN  Area = Rajura ( Maharashtra )  
1          NaN      NaN         NaN                            NaN  
2          NaN      NaN         NaN                            NaN  
3          NaN      NaN         NaN                            NaN  
4          NaN      NaN         NaN         Date of Birth Between

#### Name of Patient

For <b>Name of Patient</b> field import a dataset <b>Gender_final.xlsx</b> downloaded from <a href="">internet<a>

In [3]:
nams=pd.read_excel("DataSets/Gender_final.xlsx")
nams.head()

name gender    race
0          barjraj      m  indian
1     ramdin verma      m  indian
2  sharat chandran      m  indian
3  birender mandal      m  indian
4             amit      m  indian

Cloning namke field 

In [4]:
nms=nams["name"].to_list()
# nms

Exluding names with special charecters and names having no surnames

In [5]:
pr_nams=[]

spchr='abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ'
for i in nms:
    i=str(i)
    invali=[j for j in i if j not in spchr]
    if(len(invali)>0):
        continue
    wrds=i.split(" ")
    if(len(wrds)>1):
        i=" ".join([l.capitalize() for l in wrds])
        pr_nams.append(i)
pr_nams=np.array(pr_nams)
pr_nams

array(['Ramdin Verma', 'Sharat Chandran', 'Birender Mandal', ...,
       'Qurratul Ayn', 'Qurratul Ayn', 'Qurratul Ayn'], dtype='<U49')

Sampling 50 Names

In [6]:
names=np.random.choice(np.array(pr_nams),size=50)
names

array(['Ramesh Kumar', 'Ashok Jain', 'Muttuk Kumaran', 'Naveen Kumar',
       'Laxmi Kumari', 'Rakhi Chauhan', 'Bir Singh', 'Sweta Anand',
       'Tanvi Raheja', 'Suman Uppal', 'Qurratul Ayn', 'Lokesh Manchanda',
       'Diwan Chand', 'Iqbal Hussain', 'Mohd Mahfooz Alam', 'Aman Jeet',
       'Prem Singh Bidla', 'Puneet Chawla', 'Abdul Waajid',
       'Muttu Kumaraswami', 'Arumin Kadalan', 'Ram Lagan Singh',
       'Sinil Kumar', 'Deep Kaur', 'Sajjan Kumar', 'Dinesh Kumar',
       'Qurratul Ayn', 'Subhadra Rani', 'Satrohan Saday', 'Tania Parveen',
       'Pooja Gupta', 'Suresh Chandra', 'Rahish Kha', 'Inder Dev',
       'Harvansh Shrama', 'Piyush Singh', 'Jyoti Ram', 'Tassadul Omar',
       'Harpreet Gil', 'Pooja Rani', 'Mohd Afzal', 'Anokha Kumar Misra',
       'Sonika Sharma', 'Masheen Yadav', 'Bhuli Merutha', 'Aatif Ali',
       'Jhalla Re', 'Veepinchand Midas', 'Roshanara Begam',
       'Sunder Singh'], dtype='<U49')

Assigning 50 names to <b>Name of Patient</b> of format dataset

In [7]:
dt["Name of Patient"]=names
dt.head()

Sr. No. Patient Id Name of Patient  Adhar ID  DOB   Parent ID  Contact No.  \
0        1  RajMah001    Ramesh Kumar       NaN  NaN  RajMahP001          NaN   
1        2  RajMah002      Ashok Jain       NaN  NaN  RajMahP002          NaN   
2        3  RajMah003  Muttuk Kumaran       NaN  NaN  RajMahP003          NaN   
3        4  RajMah004    Naveen Kumar       NaN  NaN  RajMahP004          NaN   
4        5  RajMah005    Laxmi Kumari       NaN  NaN  RajMahP005          NaN   

   Address  Unnamed: 8                     Unnamed: 9  
0      NaN         NaN  Area = Rajura ( Maharashtra )  
1      NaN         NaN                            NaN  
2      NaN         NaN                            NaN  
3      NaN         NaN                            NaN  
4      NaN         NaN         Date of Birth Between

#### Contact No.

For <b>Contact No.</b> field i generated data from <a href="">internet</a>

In [8]:
mbl="""+91 61215 09667
+91 61279 00191
+91 61279 27293
+91 61279 78747
+91 72836 80352
+91 61279 58944
+91 71907 36654
+91 88413 39384
+91 71937 64748
+91 71944 90384
+91 74248 82214
+91 61279 27074
+91 61279 45788
+91 61290 29293
+91 80738 00714
+91 80795 21774
+91 73238 78035
+91 61279 82074
+91 79799 19500
+91 71961 52122
+91 62528 50834
+91 63279 17501
+91 61279 54622
+91 67488 93417
+91 80706 05631
+91 61283 42488
+91 82177 44677
+91 80679 98588
+91 61279 48225
+91 61279 99422
+91 61279 43942
+91 80729 40603
+91 61279 87833
+91 61279 98384
+91 61279 81791
+91 61283 72910
+91 61281 24129
+91 86690 95269
+91 61279 15538
+91 87093 24565
+91 89173 96452
+91 61279 57206
+91 61217 36224
+91 61279 18810
+91 788 754 1396
+91 74476 37119
+91 79313 00662
+91 61279 65206
+91 81694 10401
+91 67108 96197"""
mob=[]
for i in mbl.split("\n"):
    mob.append(i)
mob

['+91 61215 09667',
 '+91 61279 00191',
 '+91 61279 27293',
 '+91 61279 78747',
 '+91 72836 80352',
 '+91 61279 58944',
 '+91 71907 36654',
 '+91 88413 39384',
 '+91 71937 64748',
 '+91 71944 90384',
 '+91 74248 82214',
 '+91 61279 27074',
 '+91 61279 45788',
 '+91 61290 29293',
 '+91 80738 00714',
 '+91 80795 21774',
 '+91 73238 78035',
 '+91 61279 82074',
 '+91 79799 19500',
 '+91 71961 52122',
 '+91 62528 50834',
 '+91 63279 17501',
 '+91 61279 54622',
 '+91 67488 93417',
 '+91 80706 05631',
 '+91 61283 42488',
 '+91 82177 44677',
 '+91 80679 98588',
 '+91 61279 48225',
 '+91 61279 99422',
 '+91 61279 43942',
 '+91 80729 40603',
 '+91 61279 87833',
 '+91 61279 98384',
 '+91 61279 81791',
 '+91 61283 72910',
 '+91 61281 24129',
 '+91 86690 95269',
 '+91 61279 15538',
 '+91 87093 24565',
 '+91 89173 96452',
 '+91 61279 57206',
 '+91 61217 36224',
 '+91 61279 18810',
 '+91 788 754 1396',
 '+91 74476 37119',
 '+91 79313 00662',
 '+91 61279 65206',
 '+91 81694 10401',
 '+91 67108 96197']

Filling <b>Contact No.</b> field with mobiloe numbers

In [9]:
dt["Contact No."]=mob
dt.head()

Sr. No. Patient Id Name of Patient  Adhar ID  DOB   Parent ID  \
0        1  RajMah001    Ramesh Kumar       NaN  NaN  RajMahP001   
1        2  RajMah002      Ashok Jain       NaN  NaN  RajMahP002   
2        3  RajMah003  Muttuk Kumaran       NaN  NaN  RajMahP003   
3        4  RajMah004    Naveen Kumar       NaN  NaN  RajMahP004   
4        5  RajMah005    Laxmi Kumari       NaN  NaN  RajMahP005   

       Contact No.  Address  Unnamed: 8                     Unnamed: 9  
0  +91 61215 09667      NaN         NaN  Area = Rajura ( Maharashtra )  
1  +91 61279 00191      NaN         NaN                            NaN  
2  +91 61279 27293      NaN         NaN                            NaN  
3  +91 61279 78747      NaN         NaN                            NaN  
4  +91 72836 80352      NaN         NaN         Date of Birth Between

#### Adhar ID

For <b>Adhar ID</b> field, generating 50 string of 12 random integers from 0 to 9 (for 1st position it is 1 to 9)

In [10]:
import random as rnd
adhr=[str(rnd.randint(1,9))+"".join([str(rnd.randint(0,9)) for i in range(11)]) for j in range(50)]
adhar=[]
# Formatting Adhar numbers in 4 digit goups
for i in adhr:
    t=i[:4]+" "+i[4:8]+" "+i[8:]
    adhar.append(t)
adhar

['4987 6152 0524',
 '4331 7378 3502',
 '3951 1653 3794',
 '5769 6807 5286',
 '7440 8169 1152',
 '1803 6315 4496',
 '7323 8782 7427',
 '7805 3281 4883',
 '4460 2054 4468',
 '4915 1602 9267',
 '9605 2823 3347',
 '9654 7557 7613',
 '5765 5070 9735',
 '4128 6917 1759',
 '2951 6638 3758',
 '7981 6290 8412',
 '7801 6499 7163',
 '6950 4372 6501',
 '9734 2290 4122',
 '7593 1850 4116',
 '6108 2565 9426',
 '7801 2135 0055',
 '2218 5012 5801',
 '7228 4501 0217',
 '7658 9278 4760',
 '2373 1440 4025',
 '3794 2468 4282',
 '5602 0277 8465',
 '8385 8929 4506',
 '1004 8629 1061',
 '8472 3334 0502',
 '8701 5926 9807',
 '1902 8655 5729',
 '3107 4223 7454',
 '8911 7708 7302',
 '5022 2040 6274',
 '5065 0811 4730',
 '2572 8728 5924',
 '8791 2554 1714',
 '9237 4717 3930',
 '4723 0776 1678',
 '9897 1070 7493',
 '3417 0905 3122',
 '4786 6672 6124',
 '9315 0190 1160',
 '6346 6851 4453',
 '2202 7105 8168',
 '4205 3314 0825',
 '9257 2593 8060',
 '1317 7608 0159']

Filling <b>Adhar ID</b> field with formatted adhar numbers

In [11]:
dt["Adhar ID"]=adhar
dt.head()

Sr. No. Patient Id Name of Patient        Adhar ID  DOB   Parent ID  \
0        1  RajMah001    Ramesh Kumar  4987 6152 0524  NaN  RajMahP001   
1        2  RajMah002      Ashok Jain  4331 7378 3502  NaN  RajMahP002   
2        3  RajMah003  Muttuk Kumaran  3951 1653 3794  NaN  RajMahP003   
3        4  RajMah004    Naveen Kumar  5769 6807 5286  NaN  RajMahP004   
4        5  RajMah005    Laxmi Kumari  7440 8169 1152  NaN  RajMahP005   

       Contact No.  Address  Unnamed: 8                     Unnamed: 9  
0  +91 61215 09667      NaN         NaN  Area = Rajura ( Maharashtra )  
1  +91 61279 00191      NaN         NaN                            NaN  
2  +91 61279 27293      NaN         NaN                            NaN  
3  +91 61279 78747      NaN         NaN                            NaN  
4  +91 72836 80352      NaN         NaN         Date of Birth Between

Saving the updated dataset at <b>Patient Record_50.xlsx</b>

In [12]:
dt.to_excel("DataSets/Patient Record_50.xlsx",sheet_name="Patient Record",index=False)

#### DOB

##### Date of Birth Between January 2020 to December 2023 

List all dates range from <b>March 1, 2022</b> to <b>July 29, 2023</b><br>
And then sampling 50 dates

In [13]:
dates=pd.date_range(start='1/3/2022', end='7/29/2023',freq="d").astype(str).to_list()
dob=np.random.choice(dates,size=50)
dob

array(['2022-05-31', '2022-03-31', '2022-12-11', '2022-05-10',
       '2023-02-23', '2023-03-24', '2023-06-17', '2022-05-26',
       '2023-05-25', '2022-05-07', '2022-10-15', '2022-01-11',
       '2023-06-17', '2022-11-24', '2023-01-26', '2022-06-17',
       '2022-08-25', '2022-03-07', '2023-07-20', '2022-07-19',
       '2023-02-12', '2022-02-20', '2023-06-26', '2022-02-01',
       '2022-12-08', '2022-09-24', '2023-07-18', '2023-06-07',
       '2022-03-31', '2023-06-04', '2022-10-15', '2023-01-31',
       '2022-06-18', '2022-03-08', '2022-11-15', '2022-06-12',
       '2023-01-02', '2022-05-22', '2022-07-26', '2022-11-25',
       '2022-05-06', '2023-04-20', '2023-05-11', '2022-02-04',
       '2022-10-16', '2023-06-24', '2022-04-07', '2022-11-02',
       '2023-01-13', '2022-06-06'], dtype='<U10')

Filling <b>DOB</b> field with <b>dob</b> variable and removing unwanted columns

In [14]:
dt["DOB"]=dob
dt=dt[["Sr. No.","Patient Id","Name of Patient","Adhar ID","DOB","Parent ID","Contact No.","Address"]]
dt.head()

Sr. No. Patient Id Name of Patient        Adhar ID         DOB   Parent ID  \
0        1  RajMah001    Ramesh Kumar  4987 6152 0524  2022-05-31  RajMahP001   
1        2  RajMah002      Ashok Jain  4331 7378 3502  2022-03-31  RajMahP002   
2        3  RajMah003  Muttuk Kumaran  3951 1653 3794  2022-12-11  RajMahP003   
3        4  RajMah004    Naveen Kumar  5769 6807 5286  2022-05-10  RajMahP004   
4        5  RajMah005    Laxmi Kumari  7440 8169 1152  2023-02-23  RajMahP005   

       Contact No.  Address  
0  +91 61215 09667      NaN  
1  +91 61279 00191      NaN  
2  +91 61279 27293      NaN  
3  +91 61279 78747      NaN  
4  +91 72836 80352      NaN

Saving the updated dataset at <b>Patient Record_50.xlsx</b>

In [15]:
dt.to_excel("DataSets/Patient Record_50.xlsx",sheet_name="Patient Record",index=False)

Formatting <b>DOB</b> field as <b>day-month-year</b>

In [16]:
dob = dt["DOB"].to_list()
dob1=[]
for i in dob:
    dob1.append("-".join(i.split("-")[::-1]))
dob1

['31-05-2022',
 '31-03-2022',
 '11-12-2022',
 '10-05-2022',
 '23-02-2023',
 '24-03-2023',
 '17-06-2023',
 '26-05-2022',
 '25-05-2023',
 '07-05-2022',
 '15-10-2022',
 '11-01-2022',
 '17-06-2023',
 '24-11-2022',
 '26-01-2023',
 '17-06-2022',
 '25-08-2022',
 '07-03-2022',
 '20-07-2023',
 '19-07-2022',
 '12-02-2023',
 '20-02-2022',
 '26-06-2023',
 '01-02-2022',
 '08-12-2022',
 '24-09-2022',
 '18-07-2023',
 '07-06-2023',
 '31-03-2022',
 '04-06-2023',
 '15-10-2022',
 '31-01-2023',
 '18-06-2022',
 '08-03-2022',
 '15-11-2022',
 '12-06-2022',
 '02-01-2023',
 '22-05-2022',
 '26-07-2022',
 '25-11-2022',
 '06-05-2022',
 '20-04-2023',
 '11-05-2023',
 '04-02-2022',
 '16-10-2022',
 '24-06-2023',
 '07-04-2022',
 '02-11-2022',
 '13-01-2023',
 '06-06-2022']

Update <b>DOB</b> field with formatted dates

In [17]:
dt["DOB"]=dob1
dt.tail()

Sr. No. Patient Id    Name of Patient        Adhar ID         DOB  \
45       46  RajMah046          Aatif Ali  6346 6851 4453  24-06-2023   
46       47  RajMah047          Jhalla Re  2202 7105 8168  07-04-2022   
47       48  RajMah048  Veepinchand Midas  4205 3314 0825  02-11-2022   
48       49  RajMah049    Roshanara Begam  9257 2593 8060  13-01-2023   
49       50  RajMah050       Sunder Singh  1317 7608 0159  06-06-2022   

     Parent ID      Contact No.  Address  
45  RajMahP046  +91 74476 37119      NaN  
46  RajMahP047  +91 79313 00662      NaN  
47  RajMahP048  +91 61279 65206      NaN  
48  RajMahP049  +91 81694 10401      NaN  
49  RajMahP050  +91 67108 96197      NaN

Saving the updated dataset at <b>Patient Record_50.xlsx</b>

In [18]:
dt.to_excel("DataSets/Patient Record_50.xlsx",sheet_name="Patient Record",index=False)

#### Address

For <b>Address</b> field, 4 areas are randomly selected
<ol>
    <li>Rajura</li>
    <li>Aheri</li>
    <li>Gadchiroli</li>
    <li>Etapalli</li>

In [19]:
area = ["Rajura","Aheri","Gadchiroli","Etapalli"]
addr=rnd.choices(area,k=50)

Filling <b>Address</b> field with randomly sampled areas

In [20]:
dt["Address"]=addr
dt.tail()

Sr. No. Patient Id    Name of Patient        Adhar ID         DOB  \
45       46  RajMah046          Aatif Ali  6346 6851 4453  24-06-2023   
46       47  RajMah047          Jhalla Re  2202 7105 8168  07-04-2022   
47       48  RajMah048  Veepinchand Midas  4205 3314 0825  02-11-2022   
48       49  RajMah049    Roshanara Begam  9257 2593 8060  13-01-2023   
49       50  RajMah050       Sunder Singh  1317 7608 0159  06-06-2022   

     Parent ID      Contact No.     Address  
45  RajMahP046  +91 74476 37119      Rajura  
46  RajMahP047  +91 79313 00662  Gadchiroli  
47  RajMahP048  +91 61279 65206      Rajura  
48  RajMahP049  +91 81694 10401    Etapalli  
49  RajMahP050  +91 67108 96197       Aheri

Saving the updated dataset at <b>Patient Record_50.xlsx</b>

In [21]:
dt.to_excel("DataSets/Patient Record_50.xlsx",sheet_name="Patient Record",index=False)

## For sheet_2 , Measuring Parameter

import 2nd sheet dataset format

In [22]:
dt2=pd.read_excel("DataSets/Dataset Formats.xlsx",sheet_name="Measuring Parameter")
dt2.head()

Sr.No.  Time  Date Patient Id  Height (cm)  Weight (kg)  ForeArm Size (cm)  \
0       1   NaN   NaN  RajMah001          NaN          NaN                NaN   
1       2   NaN   NaN  RajMah002          NaN          NaN                NaN   
2       3   NaN   NaN  RajMah003          NaN          NaN                NaN   
3       4   NaN   NaN  RajMah004          NaN          NaN                NaN   
4       5   NaN   NaN  RajMah005          NaN          NaN                NaN   

   BMI  HeartBeat Rate  Unnamed: 9  \
0  NaN             NaN         NaN   
1  NaN             NaN         NaN   
2  NaN             NaN         NaN   
3  NaN             NaN         NaN   
4  NaN             NaN         NaN   

                                         Unnamed: 10  
0                                                NaN  
1             Fore - Arm circumference of Child has   
2                                   to be measured,   
3   Below 11.5 cm = SevereAcute Malnourishment (S...  
4  11.5 to 12.5 cm = Moderately Acute Malnourishe...

Removing unwanted columns from dataset format

In [23]:
dt2=dt2[["Sr.No.","Time","Date","Patient Id","Height (cm)","Weight (kg)","ForeArm Size (cm)","BMI","HeartBeat Rate"]]
dt2.head()

Sr.No.  Time  Date Patient Id  Height (cm)  Weight (kg)  ForeArm Size (cm)  \
0       1   NaN   NaN  RajMah001          NaN          NaN                NaN   
1       2   NaN   NaN  RajMah002          NaN          NaN                NaN   
2       3   NaN   NaN  RajMah003          NaN          NaN                NaN   
3       4   NaN   NaN  RajMah004          NaN          NaN                NaN   
4       5   NaN   NaN  RajMah005          NaN          NaN                NaN   

   BMI  HeartBeat Rate  
0  NaN             NaN  
1  NaN             NaN  
2  NaN             NaN  
3  NaN             NaN  
4  NaN             NaN

#### Date

##### Date of Birth Between September 2023 to April 2023 

List all dates range from <b>September 9, 2023</b> to <b>April 11, 2023</b><br>
And then sampling 50 dates <br>And then sort it

In [24]:
m_dats=pd.bdate_range(start="9/9/2023",end="11/4/2023").to_list()
mes_ds=pd.Series(np.random.choice(m_dats,size=50)).sort_values()
# pandas.bdate_range gives values with date and time combined with space join, but we want only date so take 0 index after splitting
mes_dates=[str(i).split(" ")[0] for i in mes_ds]

mes_dates

['2023-09-11',
 '2023-09-11',
 '2023-09-13',
 '2023-09-14',
 '2023-09-15',
 '2023-09-15',
 '2023-09-18',
 '2023-09-18',
 '2023-09-19',
 '2023-09-19',
 '2023-09-19',
 '2023-09-20',
 '2023-09-21',
 '2023-09-21',
 '2023-09-21',
 '2023-09-21',
 '2023-09-21',
 '2023-09-22',
 '2023-09-25',
 '2023-09-26',
 '2023-09-27',
 '2023-09-27',
 '2023-09-28',
 '2023-09-28',
 '2023-09-29',
 '2023-10-02',
 '2023-10-03',
 '2023-10-04',
 '2023-10-04',
 '2023-10-04',
 '2023-10-05',
 '2023-10-06',
 '2023-10-10',
 '2023-10-12',
 '2023-10-13',
 '2023-10-13',
 '2023-10-16',
 '2023-10-16',
 '2023-10-18',
 '2023-10-19',
 '2023-10-23',
 '2023-10-24',
 '2023-10-25',
 '2023-10-25',
 '2023-10-25',
 '2023-10-25',
 '2023-11-02',
 '2023-11-02',
 '2023-11-03',
 '2023-11-03']

Filling <b>Date</b> field with <b>mes_dates

In [25]:
dt2.loc[:,"Date"]=mes_dates
dt2.head()

Sr.No.  Time        Date Patient Id  Height (cm)  Weight (kg)  \
0       1   NaN  2023-09-11  RajMah001          NaN          NaN   
1       2   NaN  2023-09-11  RajMah002          NaN          NaN   
2       3   NaN  2023-09-13  RajMah003          NaN          NaN   
3       4   NaN  2023-09-14  RajMah004          NaN          NaN   
4       5   NaN  2023-09-15  RajMah005          NaN          NaN   

   ForeArm Size (cm)  BMI  HeartBeat Rate  
0                NaN  NaN             NaN  
1                NaN  NaN             NaN  
2                NaN  NaN             NaN  
3                NaN  NaN             NaN  
4                NaN  NaN             NaN

#### Time

Time of to measurement in the same day should not be same, Let's find duplicated dates

In [26]:
dat_dup=dt2["Date"].duplicated(keep="first")
dat_dup

0     False
1      True
2     False
3     False
4     False
5      True
6     False
7      True
8     False
9      True
10     True
11    False
12    False
13     True
14     True
15     True
16     True
17    False
18    False
19    False
20    False
21     True
22    False
23     True
24    False
25    False
26    False
27    False
28     True
29     True
30    False
31    False
32    False
33    False
34    False
35     True
36    False
37     True
38    False
39    False
40    False
41    False
42    False
43     True
44     True
45     True
46    False
47     True
48    False
49     True
Name: Date, dtype: bool

In [27]:
# mes_time=[]
# for i in dat_dup:
# #     if i:
#         mes_time.append("NaN")
# #     else:
# #         mes_time.append(rnd.choice(mes_tym))
# mes_time

Get a list of all times in a day differ by 1 min.  

In [28]:
mes_tym=pd.date_range(start="10:00",end="12:50",freq="1min").append(pd.date_range(start="14:15",end="15:55",freq="1min"))
# pandas.bdate_range gives values with date and time combined with space join, but we want only time so take 1 index after splitting
mes_tym=[str(i).split(" ")[1][:-3] for i in mes_tym]
# mes_tym

In [29]:
# dt2.loc[:,"Time"]=mes_time
# dt2.head()

Creating a new variable <b>mes_time</b> which have the same length of dataset(50) and assign some value to it , here i copied dat_dup.</br>

Assign sorted random sample of <b>bussiness times(var mes_tym)<b/> same number as the counted continuous True values in dat_dup

In [30]:
mes_time=dat_dup.copy()
i=1
trg=1
while (i < len(dat_dup)):
    if(dat_dup[i]):
        trg+=1
    else:
        
        tmm=pd.Series(np.random.choice(mes_tym[::10],trg)).sort_values().to_list()
        mes_time[i-trg:i]=tmm
        trg=1
    
    
    i+=1
    if(i==len(dat_dup)):
        tmm=pd.Series(np.random.choice(mes_tym[::10],trg)).sort_values().to_list()
        mes_time[i-trg:i]=tmm
mes_time


C:\Users\LENOVO\AppData\Local\Temp\ipykernel_11072\2836995929.py:10: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['11:20', '11:20']' has dtype incompatible with bool, please explicitly cast to a compatible dtype first.
  mes_time[i-trg:i]=tmm


0     11:20
1     11:20
2     11:00
3     11:10
4     14:54
5     15:24
6     14:24
7     15:04
8     11:00
9     11:10
10    15:34
11    11:20
12    10:50
13    12:00
14    12:10
15    12:30
16    14:34
17    15:04
18    14:24
19    10:10
20    10:50
21    14:24
22    11:30
23    15:14
24    10:40
25    12:20
26    14:34
27    10:50
28    12:10
29    14:44
30    11:10
31    12:20
32    11:10
33    14:24
34    10:00
35    14:44
36    10:10
37    11:10
38    14:54
39    15:04
40    11:20
41    14:54
42    10:40
43    11:20
44    14:24
45    14:34
46    12:50
47    14:24
48    15:04
49    15:14
Name: Date, dtype: object

Fill <b>Time</b> field 

In [31]:

dt2.loc[:,"Time"]=mes_time
dt2

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
0        1  11:20  2023-09-11  RajMah001          NaN          NaN   
1        2  11:20  2023-09-11  RajMah002          NaN          NaN   
2        3  11:00  2023-09-13  RajMah003          NaN          NaN   
3        4  11:10  2023-09-14  RajMah004          NaN          NaN   
4        5  14:54  2023-09-15  RajMah005          NaN          NaN   
5        6  15:24  2023-09-15  RajMah006          NaN          NaN   
6        7  14:24  2023-09-18  RajMah007          NaN          NaN   
7        8  15:04  2023-09-18  RajMah008          NaN          NaN   
8        9  11:00  2023-09-19  RajMah009          NaN          NaN   
9       10  11:10  2023-09-19  RajMah010          NaN          NaN   
10      11  15:34  2023-09-19  RajMah011          NaN          NaN   
11      12  11:20  2023-09-20  RajMah012          NaN          NaN   
12      13  10:50  2023-09-21  RajMah013          NaN          NaN   
13      14  12:00  2023-09-21  RajMah014          NaN          NaN   
14      15  12:10  2023-09-21  RajMah015          NaN          NaN   
15      16  12:30  2023-09-21  RajMah016          NaN          NaN   
16      17  14:34  2023-09-21  RajMah017          NaN          NaN   
17      18  15:04  2023-09-22  RajMah018          NaN          NaN   
18      19  14:24  2023-09-25  RajMah019          NaN          NaN   
19      20  10:10  2023-09-26  RajMah020          NaN          NaN   
20      21  10:50  2023-09-27  RajMah021          NaN          NaN   
21      22  14:24  2023-09-27  RajMah022          NaN          NaN   
22      23  11:30  2023-09-28  RajMah023          NaN          NaN   
23      24  15:14  2023-09-28  RajMah024          NaN          NaN   
24      25  10:40  2023-09-29  RajMah025          NaN          NaN   
25      26  12:20  2023-10-02  RajMah026          NaN          NaN   
26      27  14:34  2023-10-03  RajMah027          NaN          NaN   
27      28  10:50  2023-10-04  RajMah028          NaN          NaN   
28      29  12:10  2023-10-04  RajMah029          NaN          NaN   
29      30  14:44  2023-10-04  RajMah030          NaN          NaN   
30      31  11:10  2023-10-05  RajMah031          NaN          NaN   
31      32  12:20  2023-10-06  RajMah032          NaN          NaN   
32      33  11:10  2023-10-10  RajMah033          NaN          NaN   
33      34  14:24  2023-10-12  RajMah034          NaN          NaN   
34      35  10:00  2023-10-13  RajMah035          NaN          NaN   
35      36  14:44  2023-10-13  RajMah036          NaN          NaN   
36      37  10:10  2023-10-16  RajMah037          NaN          NaN   
37      38  11:10  2023-10-16  RajMah038          NaN          NaN   
38      39  14:54  2023-10-18  RajMah039          NaN          NaN   
39      40  15:04  2023-10-19  RajMah040          NaN          NaN   
40      41  11:20  2023-10-23  RajMah041          NaN          NaN   
41      42  14:54  2023-10-24  RajMah042          NaN          NaN   
42      43  10:40  2023-10-25  RajMah043          NaN          NaN   
43      44  11:20  2023-10-25  RajMah044          NaN          NaN   
44      45  14:24  2023-10-25  RajMah045          NaN          NaN   
45      46  14:34  2023-10-25  RajMah046          NaN          NaN   
46      47  12:50  2023-11-02  RajMah047          NaN          NaN   
47      48  14:24  2023-11-02  RajMah048          NaN          NaN   
48      49  15:04  2023-11-03  RajMah049          NaN          NaN   
49      50  15:14  2023-11-03  RajMah050          NaN          NaN   

    ForeArm Size (cm)  BMI  HeartBeat Rate  
0                 NaN  NaN             NaN  
1                 NaN  NaN             NaN  
2                 NaN  NaN             NaN  
3                 NaN  NaN             NaN  
4                 NaN  NaN             NaN  
5                 NaN  NaN             NaN  
6                 NaN  NaN             NaN  
7                 NaN  NaN             NaN  
8                 NaN  NaN  

Saving the updated dataset at <b>Measuring Parameter_50.xlsx</b>

In [32]:
dt2.to_excel("DataSets/Measuring Parameter_50.xlsx",sheet_name="Measuring Parameter",index=False)

#### Height, Weight

Generating random values <b>from 80.00 to 105.00 for Height</b> and <b>for Weight it is from 9.00 to 20.00</b>

In [33]:
height=[rnd.randint(8000,10500)/100 for i in range(50)]
weight=[rnd.randint(900,2000)/100 for i in range(50)]
height,weight

([85.96,
  96.06,
  100.1,
  95.85,
  95.79,
  88.02,
  83.1,
  97.89,
  82.64,
  86.79,
  99.38,
  84.22,
  91.46,
  83.75,
  103.13,
  83.04,
  92.79,
  96.58,
  83.1,
  98.97,
  95.14,
  103.72,
  83.84,
  101.4,
  83.6,
  93.61,
  92.9,
  97.58,
  81.6,
  86.54,
  91.03,
  99.56,
  86.36,
  92.92,
  95.24,
  97.97,
  80.49,
  100.29,
  89.47,
  103.5,
  87.37,
  92.59,
  90.1,
  95.54,
  101.13,
  89.79,
  103.54,
  80.84,
  104.07,
  87.85],
 [18.9,
  18.37,
  13.51,
  13.31,
  9.99,
  12.54,
  15.89,
  10.11,
  17.03,
  18.69,
  19.5,
  17.21,
  11.88,
  14.83,
  18.58,
  11.89,
  12.14,
  14.0,
  10.48,
  19.75,
  12.66,
  12.14,
  10.64,
  16.69,
  13.5,
  11.64,
  15.39,
  19.79,
  17.34,
  10.62,
  19.12,
  14.07,
  18.79,
  19.97,
  15.8,
  16.85,
  18.69,
  9.86,
  17.18,
  9.4,
  16.9,
  17.73,
  15.4,
  14.01,
  15.77,
  9.1,
  15.47,
  17.05,
  15.74,
  11.07])

Filling <b>height</b> and <b>Weight</b> field

In [34]:
dt2.loc[:,"Height (cm)"]=height
dt2.loc[:,"Weight (kg)"]=weight
dt2

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
0        1  11:20  2023-09-11  RajMah001        85.96        18.90   
1        2  11:20  2023-09-11  RajMah002        96.06        18.37   
2        3  11:00  2023-09-13  RajMah003       100.10        13.51   
3        4  11:10  2023-09-14  RajMah004        95.85        13.31   
4        5  14:54  2023-09-15  RajMah005        95.79         9.99   
5        6  15:24  2023-09-15  RajMah006        88.02        12.54   
6        7  14:24  2023-09-18  RajMah007        83.10        15.89   
7        8  15:04  2023-09-18  RajMah008        97.89        10.11   
8        9  11:00  2023-09-19  RajMah009        82.64        17.03   
9       10  11:10  2023-09-19  RajMah010        86.79        18.69   
10      11  15:34  2023-09-19  RajMah011        99.38        19.50   
11      12  11:20  2023-09-20  RajMah012        84.22        17.21   
12      13  10:50  2023-09-21  RajMah013        91.46        11.88   
13      14  12:00  2023-09-21  RajMah014        83.75        14.83   
14      15  12:10  2023-09-21  RajMah015       103.13        18.58   
15      16  12:30  2023-09-21  RajMah016        83.04        11.89   
16      17  14:34  2023-09-21  RajMah017        92.79        12.14   
17      18  15:04  2023-09-22  RajMah018        96.58        14.00   
18      19  14:24  2023-09-25  RajMah019        83.10        10.48   
19      20  10:10  2023-09-26  RajMah020        98.97        19.75   
20      21  10:50  2023-09-27  RajMah021        95.14        12.66   
21      22  14:24  2023-09-27  RajMah022       103.72        12.14   
22      23  11:30  2023-09-28  RajMah023        83.84        10.64   
23      24  15:14  2023-09-28  RajMah024       101.40        16.69   
24      25  10:40  2023-09-29  RajMah025        83.60        13.50   
25      26  12:20  2023-10-02  RajMah026        93.61        11.64   
26      27  14:34  2023-10-03  RajMah027        92.90        15.39   
27      28  10:50  2023-10-04  RajMah028        97.58        19.79   
28      29  12:10  2023-10-04  RajMah029        81.60        17.34   
29      30  14:44  2023-10-04  RajMah030        86.54        10.62   
30      31  11:10  2023-10-05  RajMah031        91.03        19.12   
31      32  12:20  2023-10-06  RajMah032        99.56        14.07   
32      33  11:10  2023-10-10  RajMah033        86.36        18.79   
33      34  14:24  2023-10-12  RajMah034        92.92        19.97   
34      35  10:00  2023-10-13  RajMah035        95.24        15.80   
35      36  14:44  2023-10-13  RajMah036        97.97        16.85   
36      37  10:10  2023-10-16  RajMah037        80.49        18.69   
37      38  11:10  2023-10-16  RajMah038       100.29         9.86   
38      39  14:54  2023-10-18  RajMah039        89.47        17.18   
39      40  15:04  2023-10-19  RajMah040       103.50         9.40   
40      41  11:20  2023-10-23  RajMah041        87.37        16.90   
41      42  14:54  2023-10-24  RajMah042        92.59        17.73   
42      43  10:40  2023-10-25  RajMah043        90.10        15.40   
43      44  11:20  2023-10-25  RajMah044        95.54        14.01   
44      45  14:24  2023-10-25  RajMah045       101.13        15.77   
45      46  14:34  2023-10-25  RajMah046        89.79         9.10   
46      47  12:50  2023-11-02  RajMah047       103.54        15.47   
47      48  14:24  2023-11-02  RajMah048        80.84        17.05   
48      49  15:04  2023-11-03  RajMah049       104.07        15.74   
49      50  15:14  2023-11-03  RajMah050        87.85        11.07   

    ForeArm Size (cm)  BMI  HeartBeat Rate  
0                 NaN  NaN             NaN  
1                 NaN  NaN             NaN  
2                 NaN  NaN             NaN  
3                 NaN  NaN             NaN  
4                 NaN  NaN             NaN  
5                 NaN  NaN             NaN  
6                 NaN  NaN             NaN  
7                 NaN  NaN             NaN  
8                 NaN  NaN  

Calculating BMI = Weight/Height**2
    

In [35]:
bmi=round(dt2["Weight (kg)"]/(dt2["Height (cm)"]/100)**2,2)
dt2.loc[:,"BMI"]=bmi
dt2.head()

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
0       1  11:20  2023-09-11  RajMah001        85.96        18.90   
1       2  11:20  2023-09-11  RajMah002        96.06        18.37   
2       3  11:00  2023-09-13  RajMah003       100.10        13.51   
3       4  11:10  2023-09-14  RajMah004        95.85        13.31   
4       5  14:54  2023-09-15  RajMah005        95.79         9.99   

   ForeArm Size (cm)    BMI  HeartBeat Rate  
0                NaN  25.58             NaN  
1                NaN  19.91             NaN  
2                NaN  13.48             NaN  
3                NaN  14.49             NaN  
4                NaN  10.89             NaN

Saving the updated dataset at <b>Measuring Parameter_50.xlsx</b>

In [36]:
dt2.to_excel("DataSets/Measuring Parameter_50.xlsx",sheet_name="Measuring Parameter",index=False)

#### 	ForeArm Size (cm)

Generating random values from 7.00 to 15.00

In [37]:
arm = [rnd.randint(700,1500)/100 for i in range(50)]
arm

[8.09,
 8.38,
 11.09,
 7.14,
 7.18,
 13.64,
 14.62,
 9.4,
 8.4,
 14.71,
 10.18,
 10.66,
 8.66,
 11.25,
 11.71,
 13.77,
 10.7,
 13.09,
 8.84,
 12.79,
 10.68,
 12.33,
 14.92,
 14.62,
 14.13,
 11.15,
 8.7,
 13.96,
 7.19,
 7.15,
 10.59,
 9.66,
 12.19,
 12.12,
 7.76,
 13.61,
 9.94,
 10.85,
 12.88,
 10.6,
 7.52,
 14.65,
 7.05,
 10.57,
 14.46,
 9.83,
 8.86,
 13.38,
 8.49,
 7.29]

Filling <b>ForeArm Size (cm)</b> field

In [38]:
dt2["ForeArm Size (cm)"]=arm
dt2.head()

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
0       1  11:20  2023-09-11  RajMah001        85.96        18.90   
1       2  11:20  2023-09-11  RajMah002        96.06        18.37   
2       3  11:00  2023-09-13  RajMah003       100.10        13.51   
3       4  11:10  2023-09-14  RajMah004        95.85        13.31   
4       5  14:54  2023-09-15  RajMah005        95.79         9.99   

   ForeArm Size (cm)    BMI  HeartBeat Rate  
0               8.09  25.58             NaN  
1               8.38  19.91             NaN  
2              11.09  13.48             NaN  
3               7.14  14.49             NaN  
4               7.18  10.89             NaN

##### Categorize Malnourishement by Forearm size
arm size  0 - 11.5cm == SAM<br>
          11.6 - 12.4cm == MAM<br>
          12.5cm - 13.4     == EU<br>
          13.5 - 15 NAM<br>


In [39]:
com = []
for i in arm:
    if(i<=11.55):
        com.append("SAM")
    elif(11.55<i and i<=12.45):
        com.append("MAM")
    elif(12.45<i and i<=13.35):
        com.append("EU")
    elif(13.35<i):
        com.append("NAN")
dt2["Category of Malnourished Based on Arm Size"]=com
dt2.tail(10)

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
40      41  11:20  2023-10-23  RajMah041        87.37        16.90   
41      42  14:54  2023-10-24  RajMah042        92.59        17.73   
42      43  10:40  2023-10-25  RajMah043        90.10        15.40   
43      44  11:20  2023-10-25  RajMah044        95.54        14.01   
44      45  14:24  2023-10-25  RajMah045       101.13        15.77   
45      46  14:34  2023-10-25  RajMah046        89.79         9.10   
46      47  12:50  2023-11-02  RajMah047       103.54        15.47   
47      48  14:24  2023-11-02  RajMah048        80.84        17.05   
48      49  15:04  2023-11-03  RajMah049       104.07        15.74   
49      50  15:14  2023-11-03  RajMah050        87.85        11.07   

    ForeArm Size (cm)    BMI  HeartBeat Rate  \
40               7.52  22.14             NaN   
41              14.65  20.68             NaN   
42               7.05  18.97             NaN   
43              10.57  15.35             NaN   
44              14.46  15.42             NaN   
45               9.83  11.29             NaN   
46               8.86  14.43             NaN   
47              13.38  26.09             NaN   
48               8.49  14.53             NaN   
49               7.29  14.34             NaN   

   Category of Malnourished Based on Arm Size  
40                                        SAM  
41                                        NAN  
42                                        SAM  
43                                        SAM  
44                                        NAN  
45                                        SAM  
46                                        SAM  
47                                        NAN  
48                                        SAM  
49                                        SAM

#### HeartBeat Rate
generating data for heart beat rate ranging from 70 to 160

In [40]:
heartBeat= [rnd.randint(70,160) for i in range(50)]

Fiiling <b>HeartBeat Rate</b>

In [41]:
dt2["HeartBeat Rate"] = heartBeat
dt2.head()

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
0       1  11:20  2023-09-11  RajMah001        85.96        18.90   
1       2  11:20  2023-09-11  RajMah002        96.06        18.37   
2       3  11:00  2023-09-13  RajMah003       100.10        13.51   
3       4  11:10  2023-09-14  RajMah004        95.85        13.31   
4       5  14:54  2023-09-15  RajMah005        95.79         9.99   

   ForeArm Size (cm)    BMI  HeartBeat Rate  \
0               8.09  25.58             159   
1               8.38  19.91              77   
2              11.09  13.48             113   
3               7.14  14.49             140   
4               7.18  10.89              98   

  Category of Malnourished Based on Arm Size  
0                                        SAM  
1                                        SAM  
2                                        SAM  
3                                        SAM  
4                                        SAM

Rearrenging the dataset

In [42]:
dt2=dt2[["Sr.No.","Time","Date","Patient Id","Height (cm)","Weight (kg)","ForeArm Size (cm)","Category of Malnourished Based on Arm Size","BMI","HeartBeat Rate"]]
dt2.head(10)# 

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
0       1  11:20  2023-09-11  RajMah001        85.96        18.90   
1       2  11:20  2023-09-11  RajMah002        96.06        18.37   
2       3  11:00  2023-09-13  RajMah003       100.10        13.51   
3       4  11:10  2023-09-14  RajMah004        95.85        13.31   
4       5  14:54  2023-09-15  RajMah005        95.79         9.99   
5       6  15:24  2023-09-15  RajMah006        88.02        12.54   
6       7  14:24  2023-09-18  RajMah007        83.10        15.89   
7       8  15:04  2023-09-18  RajMah008        97.89        10.11   
8       9  11:00  2023-09-19  RajMah009        82.64        17.03   
9      10  11:10  2023-09-19  RajMah010        86.79        18.69   

   ForeArm Size (cm) Category of Malnourished Based on Arm Size    BMI  \
0               8.09                                        SAM  25.58   
1               8.38                                        SAM  19.91   
2              11.09                                        SAM  13.48   
3               7.14                                        SAM  14.49   
4               7.18                                        SAM  10.89   
5              13.64                                        NAN  16.19   
6              14.62                                        NAN  23.01   
7               9.40                                        SAM  10.55   
8               8.40                                        SAM  24.94   
9              14.71                                        NAN  24.81   

   HeartBeat Rate  
0             159  
1              77  
2             113  
3             140  
4              98  
5              77  
6              79  
7             146  
8             140  
9             127

Saving the updated dataset at <b>Measuring Parameter_50.xlsx</b>

In [43]:
dt2.to_excel("DataSets/Measuring Parameter_50.xlsx",sheet_name="Measuring Parameter",index=False)

Formatting Date as day-month-year

In [44]:
date = dt2["Date"].to_list()
date1=[]
for i in date:
    date1.append("-".join(i.split("-")[::-1]))
dt2["Date"] = date1
dt2.tail()

Sr.No.   Time        Date Patient Id  Height (cm)  Weight (kg)  \
45      46  14:34  25-10-2023  RajMah046        89.79         9.10   
46      47  12:50  02-11-2023  RajMah047       103.54        15.47   
47      48  14:24  02-11-2023  RajMah048        80.84        17.05   
48      49  15:04  03-11-2023  RajMah049       104.07        15.74   
49      50  15:14  03-11-2023  RajMah050        87.85        11.07   

    ForeArm Size (cm) Category of Malnourished Based on Arm Size    BMI  \
45               9.83                                        SAM  11.29   
46               8.86                                        SAM  14.43   
47              13.38                                        NAN  26.09   
48               8.49                                        SAM  14.53   
49               7.29                                        SAM  14.34   

    HeartBeat Rate  
45             116  
46             159  
47             137  
48              94  
49              77

Finally Save the third sheet at <b>Measuring Parameter_50.xlsx</b>

In [45]:
dt2.to_excel("DataSets/Measuring Parameter_50.xlsx",sheet_name="Measuring Parameter",index=False)

# For Sheet_3, Result Output

Importing 3rd dataset format

In [46]:
dt3=pd.read_excel("DataSets/Dataset Formats.xlsx",sheet_name="Result Output")
dt3.head()

Sr.No. Patient Id  Time  Date  Malnourished or Not  Type of Malnourishment  \
0       1  RajMah001   NaN   NaN                  NaN                     NaN   
1       2  RajMah002   NaN   NaN                  NaN                     NaN   
2       3  RajMah003   NaN   NaN                  NaN                     NaN   
3       4  RajMah004   NaN   NaN                  NaN                     NaN   
4       5  RajMah005   NaN   NaN                  NaN                     NaN   

   Precautionary Measures Unnamed: 7  Unnamed: 8              Unnamed: 9  \
0                     NaN        NaN         NaN                     NaN   
1                     NaN        NaN         NaN                     NaN   
2                     NaN        NaN         NaN                     NaN   
3                     NaN  Category          NaN  Type of Malnourishment   
4                     NaN        NaN         NaN                     NaN   

  Unnamed: 10  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN

Removing unwanted fields

In [47]:
dt3=dt3[["Sr.No.","Patient Id","Time","Date","Malnourished or Not","Type of Malnourishment","Precautionary Measures"]]
dt3.head()

Sr.No. Patient Id  Time  Date  Malnourished or Not  Type of Malnourishment  \
0       1  RajMah001   NaN   NaN                  NaN                     NaN   
1       2  RajMah002   NaN   NaN                  NaN                     NaN   
2       3  RajMah003   NaN   NaN                  NaN                     NaN   
3       4  RajMah004   NaN   NaN                  NaN                     NaN   
4       5  RajMah005   NaN   NaN                  NaN                     NaN   

   Precautionary Measures  
0                     NaN  
1                     NaN  
2                     NaN  
3                     NaN  
4                     NaN

#### Time, Date
copying <b>Date</b> and <b>Time</b> field from 2nd sheet

In [48]:
dt3["Time"] = dt2["Time"]
dt3["Date"] = dt2["Date"]
dt3.head()

Sr.No. Patient Id   Time        Date  Malnourished or Not  \
0       1  RajMah001  11:20  11-09-2023                  NaN   
1       2  RajMah002  11:20  11-09-2023                  NaN   
2       3  RajMah003  11:00  13-09-2023                  NaN   
3       4  RajMah004  11:10  14-09-2023                  NaN   
4       5  RajMah005  14:54  15-09-2023                  NaN   

   Type of Malnourishment  Precautionary Measures  
0                     NaN                     NaN  
1                     NaN                     NaN  
2                     NaN                     NaN  
3                     NaN                     NaN  
4                     NaN                     NaN

#### Malnourished or Not
If one has <b>BMI</b> greater than 18.5 the he/she is not Malnourished otherwise Not</br>
Making the field

In [49]:
malnr=dt2["BMI"]<=18.5
malnr=["Yes" if i else "No" for i in malnr]
dt3["Malnourished or Not"]=malnr


dt3.tail()

Sr.No. Patient Id   Time        Date Malnourished or Not  \
45      46  RajMah046  14:34  25-10-2023                 Yes   
46      47  RajMah047  12:50  02-11-2023                 Yes   
47      48  RajMah048  14:24  02-11-2023                  No   
48      49  RajMah049  15:04  03-11-2023                 Yes   
49      50  RajMah050  15:14  03-11-2023                 Yes   

    Type of Malnourishment  Precautionary Measures  
45                     NaN                     NaN  
46                     NaN                     NaN  
47                     NaN                     NaN  
48                     NaN                     NaN  
49                     NaN                     NaN

#### Type of Malnourishment
Categorizing type of Malnourishment accoding to following conditions

In [50]:
bmi = dt2["BMI"].to_list()
tm=[]
for i in bmi:
    if(i< 18.5):
        tm.append("Under Weight")
    elif(18.5 <=i and i<24.9):
        tm.append("Normal")
    elif(24.9 <=i and i <29.9):
        tm.append("Over weight")
    elif(29.9 <= i and i<35):
        tm.append("Obesity")
dt3["Type of Malnourishment"]=tm
dt3.head()

Sr.No. Patient Id   Time        Date Malnourished or Not  \
0       1  RajMah001  11:20  11-09-2023                  No   
1       2  RajMah002  11:20  11-09-2023                  No   
2       3  RajMah003  11:00  13-09-2023                 Yes   
3       4  RajMah004  11:10  14-09-2023                 Yes   
4       5  RajMah005  14:54  15-09-2023                 Yes   

  Type of Malnourishment  Precautionary Measures  
0            Over weight                     NaN  
1                 Normal                     NaN  
2           Under Weight                     NaN  
3           Under Weight                     NaN  
4           Under Weight                     NaN

Saving the dataset at 

In [51]:
dt3.to_excel("DataSets/Result Output_50.xlsx",sheet_name="Result Output",index=False)

#### Precautionary Measures

In [52]:
######

Finally  save the updated dataset

In [53]:
dt3.to_excel("DataSets/Result Output_50.xlsx",sheet_name="Result Output",index=False)

Now we have 3 files
<ol>
    <li>Patient Record_50.xlsx</li>
    <li>Measuring Parameter_50.xlsx</li>
    <li>Result Output_50.xlsx</li>
</ol>
at "./DataSets"  are required datasets